### ISSUE TITLE
No parent of path 

### ISSUE THREAD

**Issue by cliffvick** [probablykasper/time-machine-inspector#4]
When launching the latest version (1.1.2) I get the following error, "No parent of path".  I haven't tried to build and run this yet, just downloaded the latest release and tried to run it.
![image](https://user-images.githubusercontent.com/72509/168704362-c441be51-91c9-48ab-ad2c-c5db3afcd340.png)

**Comment by probablykasper** [comment]
I'll improve that error message a bit in hopes of that helping.

The way the app works is it reads the output from a `tmutil` command, and puts every path into a map that basically looks like this (example):
```ts
DirMap = {
  '/': {
    'Applications': null,
    'Users': null,
  },
  '/Users': {
    'user1': null,
  },
}
```
So to add `/myfiles/documents/file.txt`, it needs to
- add `file.txt` inside `/myfiles/documents`
- add `documents` inside `/myfiles`
- add `myfiles` inside `/`

So basically to add `/myfiles/documents`, it needs to get `/myfiles` from that path, and that's what's failing in this case

**Comment by cliffvick** [comment]
On the system that I tried running this on, my timemachine disk is external - could this be what's causing this problem?

What tmutil command is it running?

**Comment by cliffvick** [comment]
I've had a moment to look at the source and I see that the tmutil command that is being executed at startup is `tmutil listbackups`. When I execute this on my machine, I don't get paths, but rather I get a list like the following:
```
2020-06-17-115149
2020-06-24-183416
2020-07-01-191047
2020-07-08-083741
2020-07-15-193333
2020-07-22-093141
2020-07-29-003243
2020-08-04-133727
2020-08-11-033558
2020-08-17-183318
2020-08-24-083446
2020-08-30-233236
```

**Comment by probablykasper** [comment]
Oh weird, for me it returns actual paths:
```
/Volumes/Time Machine Backups/Backups.backupdb/kasper-imac-2019/2022-03-20-205253
/Volumes/Time Machine Backups/Backups.backupdb/kasper-imac-2019/2022-03-27-222610
etc
```

**Comment by probablykasper** [comment]
If you run `man tmutil`, what does it say for `listbackups`? For me it says that it should print paths:
```
     listbackups
             Print paths for all of this computer's completed snapshots.
```

**Comment by cliffvick** [comment]
yes - that's exactly what it says.  I've been looking for a bit online to see if anyone has mentioned this, but I haven't found anything yet.  It might be an OS difference -- I'm currently running 11.3 (Big Sur) and haven't updated yet, so that might have something to do with it?

I discovered that `tmutil machinedirectory` will give you the actual path where the names can be appended.

I'm just starting to learn rust, but I expect that it wouldn't be that difficult to check if the first path returned is an actual path, and, if not, call `tmutil machinedirectory` to get the prefix and then that could be appended to the results of `tmutil listbackups` (what a PITA).

I'd really rather find out why `tmutil` doesn't return paths on my machine - any ideas?

**Comment by cliffvick** [comment]
Is your `/Volumes/Time Machine Backups` a local or external drive? That could be the difference.

**Comment by probablykasper** [comment]
It's a network connected AirPort drive (or whatever it's called). I'm also on macOS 10.15, so that could be the difference too

**Comment by cliffvick** [comment]
Ok, well, that's two differences: 1) Different OS (10.15 vs. 11.3), and 2) network connected AirPort vs. external drive connected via USB (or thunderbolt - I have no idea it's been so long since I plugged that drive in).

In any case, I'm just learning Rust (I've developed in several other languages for a while), and I stumbled onto Tauri recently and this was the first app that I downloaded to try it out.

If you don't have time to deal with this (or don't want to), I'd be happy to try to make it work for both cases, but it might take me a little while to get up to speed on Rust, but I'd be interested in making the attempt :)

**Comment by probablykasper** [comment]
Would be great if could! The output just needs to be turned into full paths (they have to be full paths because that's needed for running the `tmutil compare` command afaik)

**Comment by cliffvick** [comment]
OK! I'll give it a shot :)

**Comment by probablykasper** [comment]
Should be fixed in [v1.1.5](https://github.com/probablykasper/time-machine-inspector/releases/tag/v1.1.5)

**Comment by AddictArts** [comment]
I'll try and have a look at what is causing it, but it is happening still.

Mac OS: 11.6.8

**Comment by probablykasper** [comment]
Should be fixed in [v1.2.0](https://github.com/probablykasper/time-machine-inspector/releases/tag/v1.2.0)
